packages/common-ui-web/src/components/VehicleTypeSelection/VehicleTypeSelection.tsx

```diff
@@ -73,32 +97,67 @@ export const VehicleTypeSelection = i18nWrap(
     const { rootStyles } = useMonkTheme();
     const sliderRef = useRef<HTMLDivElement>(null);
 
+    const onValidate = (type: VehicleType) => {
+      if (patchInspection && inspectionId) {
+        updateInspectionVehicule({ inspectionId, vehicle: { type } });
+      }
+      onSelectVehicleType?.(type);
+    };
+
+    useEffect(() => {
+      const fetchInspection = async () => {
+        if (patchInspection && inspectionId) {
+          const inspection = await getInspection({
+            id: inspectionId,
+          });
+
+          const vehicleTypeFoundInInspection = Object.values(VehicleType).find(
+            (vehicleType) => vehicleType === inspection.body.vehicle?.vehicle_type,
```

Comment:
instead of using the response body, use the object in the result of the request in the entities